user-service/src/test/java/com/smoothstack/utopia/IntegrationTests.java

```diff
@@ -0,0 +1,199 @@
+import org.springframework.test.web.servlet.MockMvc;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestMethodOrder(OrderAnnotation.class)
+@TestPropertySource(locations = { "classpath:h2-database.properties" })
+public class IntegrationTests {
+  
+  @Autowired
+  MockMvc mockMvc;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  @DisplayName("test update user by id, expect 409 conflict email already taken")
+  @Test
+  @Order(1)
+  void testUpdateByIdEmailConflict() throws Exception {
+    User user = new User(4L, "newFirst", "newLast", "password", "user@1.com", 0);
+    String userJson = objectMapper.writeValueAsString(user);
+    mockMvc.perform(
+      put("/users")
+        .content(userJson)
+        .contentType(MediaType.APPLICATION_JSON)
+    )
+      .andExpect(status().isConflict());
+  }
+  
+  @DisplayName("test update user by id change role, expect unauthorized")
+  @Test
+  @Order(2)
+  void testUpdateByIdChangeRoleUnauthorized() throws Exception {
+    User user = new User(4L, "newFirst", "newLast", "password", "user@0.com", 1);
+    String userJson = objectMapper.writeValueAsString(user);
+    mockMvc.perform(
+      put("/users")
+        .content(userJson)
+        .contentType(MediaType.APPLICATION_JSON)
+    )
+      .andExpect(status().isUnauthorized());
+  }
+  
+  @DisplayName("test update user by id, expect 200 ok")
+  @Test
+  @Order(3)
+  void testUpdateById() throws Exception {
+    User user = new User(4L, "newFirst", "newLast", "password", "user@4.com", 0);
+    String userJson = objectMapper.writeValueAsString(user);
+    mockMvc.perform(
+      put("/users")
+        .content(userJson)
+        .contentType(MediaType.APPLICATION_JSON)
+    )
+      .andExpect(status().isOk())
+      .andExpect(jsonPath("$.firstName", is("newFirst")))
+      .andExpect(jsonPath("$.lastName", is("newLast")))
+      .andExpect(jsonPath("$.password", is("password")));
+  }
+
+  @DisplayName("test delete user by id, expect 200 ok")
+  @Test
+  @Order(4)
+  void testDeleteById() throws Exception {
+    mockMvc.perform(delete("/users/4"))
+      .andExpect(status().isOk());
+    mockMvc.perform(get("/users?findall=true"))
+      .andExpect(jsonPath("$", hasSize(3)));
+    mockMvc.perform(get("/users/4"))
+      .andExpect(status().isNotFound());
+  }
+  
+  @DisplayName("test delete user by id, expect 404 not found")
+  @Test
+  void testDeleteByIdNotFound() throws Exception {
+    mockMvc.perform(delete("/users/0"))
+      .andExpect(status().isBadRequest());
+  }
+
+  @DisplayName("test find all users, expect 200, json response, correct size")
+  @Test
+  void testFindAll() throws Exception {
+    mockMvc.perform(get("/users?findall=true"))
+    .andExpect(status().isOk())
+    .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+    .andExpect(jsonPath("$", hasSize(3)));
+  }
+  
+  @DisplayName("test find by user id admin search, expect 200")
+  @Test
+  void testAdminFindById() throws Exception {
```

Comment:
Why is this an "admin" search and not a "RequestParam" test?

Reply:
I named it that way because the method I am testing in `UserController.java` is called `adminUserSearch`. It is only intended for use by a user that has the `ADMIN` role.